core/math/projection.h

```diff
@@ -34,6 +34,8 @@
 #include "core/math/vector3.h"
 #include "core/math/vector4.h"
 
+struct Quaternion;
+
```

Comment:
don't forward declare Quaternion.

As I mention elsewhere, I would use Plane for this since it literally is used to refer to a Plane, but if Plane is somehow not working, use a Vector4. Don't use Quaternion.